src/frontends/pytorch/src/op/trilu.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+namespace base {
+
+OutputVector translate_base_triu_tril(NodeContext& context, bool upper) {
+    auto input_tensor = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<opset10::ShapeOf>(input_tensor));
+    auto zero = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {0}));
+    auto one = context.mark_node(opset10::Constant::create(element::i64, Shape{}, {1}));
+    auto minus_one = context.mark_node(opset10::Constant::create(element::i32, Shape{}, {-1}));
+    auto minus_two = context.mark_node(opset10::Constant::create(element::i32, Shape{}, {-2}));
+    const auto m = context.mark_node(std::make_shared<opset10::Gather>(input_shape, minus_one, zero));
```

Comment:
Is it necessary to mix `i32` and `i64` types here?

Reply:
aligned